### ISSUE TITLE
Problem with warm start

### ISSUE THREAD

**Issue by sebastianfastert** [EpistasisLab/tpot#1026]
I'm trying to plot the score of the best pipeline after every generation. I tried to do that with warm start. The problem is that, if I understood everything correctly, the following two examples should be identical.

I know that the number of generations and population are too low. I'm using these just for testing.

Thanks for your work and help!

Setup:
```
iris = load_iris()
X_train, X_test, y_train, y_test = train_test_split(iris.data.astype(np.float64),
    iris.target.astype(np.float64), train_size=0.75, test_size=0.25, random_state=42)
```
First version:
```
tpot = TPOTClassifier(generations=5, population_size=2, verbosity=2, random_state=42, warm_start=True)
tpot.fit(X_train, y_train)
```
Second version:
```
tpot2 = TPOTClassifier(generations=1, population_size=2, verbosity=2, random_state=42, warm_start=True)
for _ in range(0, 5):
    tpot2.fit(X_train, y_train)
```

## Current result
The first version outputs these CV scores:
Generation 1 - Current best internal CV score: 0.9462450592885375
Generation 2 - Current best internal CV score: 0.9462450592885375
Generation 3 - Current best internal CV score: 0.9462450592885375
Generation 4 - Current best internal CV score: 0.9462450592885375
Generation 5 - Current best internal CV score: 0.9545454545454547  <--

The second example these CV scores:
Generation 1 - Current best internal CV score: 0.9462450592885375
Generation 1 - Current best internal CV score: 0.9462450592885375
Generation 1 - Current best internal CV score: 0.9462450592885375
Generation 1 - Current best internal CV score: 0.9462450592885375
Generation 1 - Current best internal CV score: 0.9462450592885375    <--

**Comment by sebastianfastert** [comment]
I'm using TPOT 0.11.1

**Comment by weixuanfu** [comment]
Hmm, I can reproduce this issue in a new environment but I am not sure why it happened. A related ideas: 

_The processes in above 2 ways are not the same. For the 1st version, TPOT should run 6 generations (generation 0 is the initial population with 2 randomly generated individuals) so that [evaluation step](https://github.com/EpistasisLab/tpot/blob/master/tpot/gp_deap.py#L250) should trigger 6 times. But for 2nd version, except 1st iteration on the loop, TPOT should use the last generation after selection from pervious iteration as initial population and then run another generations after it. So the 2nd version should run 10 generations and trigger evaluation 10 times. But I don't know if this is related to random seed status in TPOT. _ 

 I am out of office these days and will look into it once I get a chance. Please let me know if you have any ideas.

**Comment by sebastianfastert** [comment]
Thanks for the quick answer. I'll try to look into it, but i have no experience...

I'm trying to plot the cv scores after every generation for my bachelorthesis. Is there some other way to get these values? I'm currently extracting them from the output...

**Comment by weixuanfu** [comment]
Hmm, so far, there is not a easy way for getting those values during one TPOT run. TPOT did not store those best scores every generation. Maybe it is a nice new feature.

**Comment by sebastianfastert** [comment]
I looked at the code and I'm going to try my best, but I'm not sure if I can find a solution.

**Comment by gosuto-inzasheru** [comment]
You could make use of the checkpoints feature and scrape it out of the comment for each step.

In the source code that is the [`pareto_front_pipeline_score`](https://github.com/EpistasisLab/tpot/blob/6efc21e4a5b3ab7623fc1ba6508f3df7449f582e/tpot/base.py#L1020) variable.
